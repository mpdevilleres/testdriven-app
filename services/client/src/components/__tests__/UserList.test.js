import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import UsersList from '../UserList';

const users = [
    {
        active: true,
        email: 'marc@gmail.com',
        id: 1,
        username: 'marc'
    },
    {
        active: true,
        email: 'philippe@gmail.com',
        id: 2,
        username: 'philippe'
    }
];

test('UsersList renders properly', () => {
    const wrapper = shallow(<UsersList users={users}/>);
    const element = wrapper.find('h4');
    expect(element.length).toBe(2);
    expect(element.get(0).props.children).toBe('marc');
});

test('UsersList renders a snapshot properly', () => {
    const tree = renderer.create(<UsersList users={users}/>).toJSON();
    expect(tree).toMatchSnapshot();
});