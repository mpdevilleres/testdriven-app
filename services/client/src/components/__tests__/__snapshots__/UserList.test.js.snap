// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
Array [
  <h4
    className="box title is-4"
  >
    marc
  </h4>,
  <h4
    className="box title is-4"
  >
    philippe
  </h4>,
]
`;
